speechbrain/utils/checkpoints.py

```diff
@@ -561,14 +561,20 @@ def save_checkpoint(
         Checkpoint
             namedtuple [see above], the saved checkpoint.
         """
-        if name is None:
-            ckpt_dir = self._new_checkpoint_dirpath()
-        else:
-            ckpt_dir = self._custom_checkpoint_dirpath(name)
-        os.makedirs(ckpt_dir)  # May raise FileExistsError, let it.
-        saved_meta = self._save_checkpoint_metafile(
-            ckpt_dir / METAFNAME, meta, end_of_epoch
-        )
+        ckpt_dir = None
+        if if_main_process():
+            if name is None:
+                ckpt_dir = self._new_checkpoint_dirpath()
+            else:
+                ckpt_dir = self._custom_checkpoint_dirpath(name)
+            os.makedirs(ckpt_dir)  # May raise FileExistsError, let it.
+            saved_meta = self._save_checkpoint_metafile(
+                ckpt_dir / METAFNAME, meta, end_of_epoch
+            )
+
+        # Communicate ckpt_dir to all procs
+        torch.distributed.broadcast_object_list([ckpt_dir], src=0)
```

Comment:
I'm a little unsure if this works when using multiple machines, perhaps we would need to use the concept of process groups and only communicate within the group somehow.